src/techui_builder/generate.py

```diff
@@ -59,16 +60,80 @@ def _get_screen_dimensions(self, file: str) -> tuple[int, int]:
 
         return (height, width)
 
+    def _get_screen_dimensions_object(
+        self, object: EmbeddedDisplay | ActionButton
+    ) -> tuple[int, int]:
+        """
+        Parses the bob files for information on the height
+        and width of the screen
+        """
+        # Read the bob file
+        root: etree._Element = etree.fromstring(str(object), None)
+        height_element: etree._Element | None = root.find("height", namespaces=None)
+        if height_element is not None:
+            height = (
+                self.default_size if (val := height_element.text) is None else int(val)
+            )
+        else:
+            height = self.default_size
+            assert "Could not obtain the size of the widget"
+
+        width_element: etree._Element | None = root.find("width", namespaces=None)
+        if width_element is not None:
+            width = (
+                self.default_size if (val := width_element.text) is None else int(val)
+            )
+        else:
+            width = self.default_size
+            assert "Could not obtain the size of the widget"
+
+        return (height, width)
+
+    def _get_screen_position_object(
+        self, object: EmbeddedDisplay | ActionButton
+    ) -> tuple[int, int]:
+        """
+        Parses the bob files for information on the y
+        and x of the screen
+        """
+        # Read the bob file
+        root: etree._Element = etree.fromstring(str(object), None)
+        y_element: etree._Element | None = root.find("y", namespaces=None)
+        if y_element is not None:
+            y = self.default_size if (val := y_element.text) is None else int(val)
+        else:
+            y = self.default_size
+            assert "Could not obtain the size of the widget"
+
+        x_element: etree._Element | None = root.find("x", namespaces=None)
+        if x_element is not None:
+            x = self.default_size if (val := x_element.text) is None else int(val)
+        else:
+            x = self.default_size
+            assert "Could not obtain the size of the widget"
+
+        return (y, x)
+
     # Make groups
     def _get_group_dimensions(self, widget_list: list[EmbeddedDisplay | ActionButton]):
         """
         Takes in a list of widget screens and finds the
```

Comment:
Is "screens" needed here?
```suggestion
        Takes in a list of widgets and finds the
```